docs/proposals/diagnose-experience.md

```diff
@@ -0,0 +1,258 @@
+> Request method: "GET"
+> Request headers:
+   "User-Agent": "oras/1.2.0+Homebrew" 
+DEBU[0002] Response #1
+< Response Status: "200 OK"
+< Response headers:
+   "Content-Type": "application/json"
+   "Docker-Distribution-Api-Version": "registry/2.0"
+   "Date": "Fri, 02 Aug 2024 16:08:05 GMT"
+   "Content-Length": "69"
+   "X-Github-Request-Id": "9FC6:30019C:17C0D:1C46C:66AD0464" 
+```
+
+**Suggested debug logs:**
+
+```
+2024-08-02 23:56:02 > msg=Request #0
+> Request URL: "https://ghcr.io/v2/oras-project/oras/manifests/v1.2.0"
+> Request method: "GET"
+> Request headers:
+   "User-Agent": "oras/1.2.0+Homebrew"
+   "Accept": "application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, application/vnd.oci.artifact.manifest.v1+json" 
+
+
+2024-08-02 23:56:03 <  msg=Response #0
+< Response Status: "401 Unauthorized"
+< Response headers:
+   "Content-Length": "73"
+   "X-Github-Request-Id": "9FC6:30019C:17C06:1C462:66AD0463"
+   "Content-Type": "application/json"
+   "Www-Authenticate": "Bearer realm=\"https://ghcr.io/token\",service=\"ghcr.io\",scope=\"repository:oras-project/oras:pull\""
+   "Date": "Fri, 02 Aug 2024 23:56:03 GMT"
+< Response body:
+{
+        "errors": [
+            {
+                "code": "<UNAUTHORIZED>",
+                "message": "<message describing condition>",
+                "detail": "<unstructured>"
+            },
+            ...
+        ]
+    }
+
+2024-08-02 23:56:02 > msg=Request #1
+> Request URL: "https://ghcr.io/token?scope=repository%3Aoras-project%2Foras%3Apull&service=ghcr.io"
+> Request method: "GET"
+> Request headers:
+   "User-Agent": "oras/1.2.0+Homebrew" 
+
+
+
+2024-08-02 23:56:03 < msg=Response #1
+< Response Status: "200 OK"
+< Response headers:
+   "Content-Type": "application/json"
+   "Docker-Distribution-Api-Version": "registry/2.0"
+   "Date": "Fri, 02 Aug 2024 16:08:05 GMT"
+   "Content-Length": "69"
+   "X-Github-Request-Id": "9FC6:30019C:17C0D:1C46C:66AD0464" 
+<  Response body:
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:42c524c48e0672568dbd2842d3a0cb34a415347145ee9fe1c8abaf65e7455b46",
+      "size": 1239,
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      }
+    },
+    ···
+}
+
+
+```
+
+### Show user's environment details
+
+Output the user's running environment details of ORAS such as operating system and architecture information would be helpful to help the ORAS developers locate the issue and reproduce easier. 
+
+For example, the operating system and architecture are supposed to be outputted in `oras version`: 
+
+```bash
+$ oras version
+
+ORAS Version:    1.2.0+Homebrew
+Go version: go1.22.3
+OS/Arch: Linux AMD64
```

Comment:
It is better to use the standard os/arch format: `linux/amd64` (i.e. `<OS>/<ARCH>`).

Reply:
Example:
```console
$ docker version
Client:
 Version:           27.2.0
 API version:       1.47
 Go version:        go1.21.13
 Git commit:        3ab4256
 Built:             Tue Aug 27 14:14:20 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Desktop  ()
 Engine:
  Version:          27.2.0
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.21.13
  Git commit:       3ab5c7d
  Built:            Tue Aug 27 14:15:15 2024
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.7.20
  GitCommit:        8fc6bcff51318944179630522a095cc9dbf9f353
 runc:
  Version:          1.1.13
  GitCommit:        v1.1.13-0-g58aa920
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

Reply:
Updated to `<OS>/<ARCH>`